portfolio/src/main/java/com/google/sps/servlets/RandomFacts.java

```diff
@@ -0,0 +1,30 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+
+/** Handles requests sent to the /hello URL. Try running a server and navigating to /hello! */
+@WebServlet("/random-facts")
```

Comment:
This servlet functions in exactly the same way as the button previously did. I implemented it to practice the json concepts in the week 2 walkthrough.

Reply:
nit: instead of sending all of the facts to your javascript code, you could just pick a random fact in your Java code and return it from this HTTP call.